x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecycleType.java

```diff
@@ -64,11 +64,16 @@
     /**
      * validates whether the specified <code>phases</code> are valid for this
      * policy instance.
-     * 
+     *
      * @param phases
      *            the phases to verify validity against
      * @throws IllegalArgumentException
      *             if a specific phase or lack of a specific phase is invalid.
      */
     void validate(Collection<Phase> phases);
+
+    /**
+     * @return this type's valid actions and their step creators
+     */
+    Map<String, StepsFactory.ActionStepsCreator<?>> getActionStepsCreators();
```

Comment:
This can't be a method on `LifecycleType` because the Lifecycle type needs to be in the API but the `ActionStepCreator`s cannot be. I think we should have `ActionStepCreator` registered globally and not tied to a `LifecycleType` and then each `LifecycleType` can choose whether it supports each Action

Reply:
ah, I forgot the type param for policies. you're right. will see about moving it. I just wanted each type to be responsible for supplying its own actions